### PR TITLE
PR: Remove Kite completion provider code

### PR DESCRIPTION
**Pull request by jsbautista**
## Description of Changes
Remove kite from the Spyder main repository

Related to https://github.com/spyder-ide/spyder-kite/pull/1

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
@jsbautista

### CODE REVIEW

**Review by dalthviz — COMMENTED** [review]
Thanks for the work here @jsbautista ! I left a comment regarding another entry to remove from the `NOTICE.txt` related with `pybloom_pyqt` utils code (which also I missing to be removed).

Besides that, I think we need to update the docstring for the completions plugin at: 
* https://github.com/jsbautista/spyder/blob/f79aafd044be814db8b77985188d8c15fb8a59e7/spyder/plugins/completion/plugin.py#L67 (remove kite mention)
* https://github.com/jsbautista/spyder/blob/f79aafd044be814db8b77985188d8c15fb8a59e7/spyder/plugins/completion/plugin.py#L1234 (remove kite mention and just put providers in general)

Also, I'm not totally sure what we should do with the entry point validation for Kite at https://github.com/jsbautista/spyder/blob/f79aafd044be814db8b77985188d8c15fb8a59e7/spyder/plugins/completion/plugin.py#L239-L244

What do you think about that last point @ccordoba12 ? Should we keep the entry point validation for Kite?

File: NOTICE.txt [hunk 1]
```diff
@@ -1479,37 +1479,6 @@ spyder/images/light/file_type_tex.svg
...
-
-spyder/images/dark/kite.svg
-spyder/images/kite/kite_copilot.png
-spyder/images/light/kite.svg
-
-
-===============================================================================
-
-
-
```
[the remark this answers is not in the document]
Comment (dalthviz) [reply NOTICE.txt:1512]: Also, the related entry for the `pybloom_pyqt` files in the `modules_test` script at  https://github.com/jsbautista/spyder/blob/f79aafd044be814db8b77985188d8c15fb8a59e7/.github/scripts/modules_test.sh#L118-L120 should be removed

File: spyder/plugins/completion/plugin.py [hunk 2]
```diff
@@ -1231,7 +1231,7 @@ def match_and_reply(self, req_id: int):
         else:
             # Any empty response will be discarded and the completion
             # loop will wait for the next non-empty response.
-            # This should fix the scenario where Kite does not have a
+            # This should fix the scenario where Providers does not have a
```
Comment (dalthviz) [inline spyder/plugins/completion/plugin.py:1234]:
```suggestion
            # This should fix the scenario where a providers does not have a
```

**Review by dalthviz — APPROVED** [review]
Thanks @jsbautista ! I did a small suggestion for text inside a comment and I think the only missing element to check is what to do about the entry point validation for Kite at https://github.com/jsbautista/spyder/blob/f79aafd044be814db8b77985188d8c15fb8a59e7/spyder/plugins/completion/plugin.py#L239-L244

What do you think about that @ccordoba12 ?

Other than that this LGTM :+1:

**Comment by ccordoba12** [comment]
At first glance, this looks good to me. Thanks @jsbautista for your work on it!

The only thing missing is to allow the external Kite code completion plugin to be loaded in Spyder. I don't remember where that's forbidden right now, but you can take a look at it with @dalthviz.

**Review by ccordoba12 — APPROVED** [review]
Thanks @jsbautista and @dalthviz for your work on this!

**Pull request merged**